src/sentry/rules/conditions/base.py

```diff
@@ -1,12 +1,19 @@
 import abc
 from collections.abc import Sequence
 from datetime import datetime
+from typing import TypedDict
 
 from sentry.eventstore.models import GroupEvent
 from sentry.rules.base import EventState, RuleBase
 from sentry.types.condition_activity import ConditionActivity
 
 
+class GenericCondition(TypedDict):
```

Comment:
This is technically shared with all filter classes as well, although there's no immediate need to create filter typeddicts so just gonna leave this as Condition only for now